src/main/java/org/opentestsystem/delivery/logging/EventLogger.java

```diff
@@ -7,31 +7,149 @@
+
+  private static final Logger logger = LoggerFactory.getLogger(EventLogger.class);
+  private static final Marker marker = MarkerFactory.getMarker("event-logger");
+  private static final ObjectWriter writer = new ObjectMapper().writer();
+  private static Map<ILogEvent, Long> eventTimes = new HashMap<>();
+
+  public enum Checkpoint {
+    NONE,
+    ENTER,
+    EXIT,
+  }
+
+  public interface ILogEvent {
+    String name();
+  }
+
+  public interface IEventData {
+    String name();
+  }
+
+  public enum LogEvent implements ILogEvent {
+    LOGIN,
+    LOGOUT,
+    UNKNOWN,
+  }
+
+  public enum EventData implements IEventData {
+    APP,
+    ELAPSED_TIME,
+    CHECKPOINT,
+    SESSION_ID,
+    SUB_EVENT,
+  }
+
+  /**
+   * Log event occurrences to centralized logging.
+   * <p>
+   * These events are intended to be sent to a logstash server and searchable using elastic search.
+   *
+   * @param app  name of the app writing the log
+   * @param logEvent  name of the event occurring
+   * @param checkpoint  an optional checkpoint name
+   * @param sessionId  an optional unique identifier common to a set of log entries (generally a request's session ID)
+   * @param message  an optional free-format message string to add to the log entry
+   * @param data optional searchable event data to add to the recorded event
+   */
+  public static void info(final String app, final ILogEvent logEvent, final String checkpoint, final String sessionId,
+                          final String message, final Map<IEventData, Object> data) {
+    log(formatMessage(logEvent, message), getFieldMap(app, logEvent, checkpoint, sessionId, data));
+  }
+
+  /**
+   * Log event errors to centralized logging.
+   * <p>
+   * These events are intended to be sent to a logstash server and searchable using elastic search.
+   *
+   * @param app  name of the app writing the log
+   * @param logEvent  name of the event occurring
+   * @param checkpoint  an optional checkpoint name
+   * @param sessionId  an optional unique identifier common to a set of log entries (generally a request's session ID)
+   * @param message  an optional free-format message string to add to the log entry
+   * @param data optional searchable event data to add to the recorded event
+   */
+  public static void error(final String app, final ILogEvent logEvent, final String checkpoint, final String sessionId,
+                           String message, final Map<IEventData, Object> data, final Exception e) {
+    error(formatMessage(logEvent, message), getFieldMap(app, logEvent, checkpoint, sessionId, data), e);
+  }
+
+  private static void log(final String event, final Map<String, Object> fields) {
+    try {
+      logger.info(marker, format("EVENT:%s JSON:({\"event_data\": %s})", event, writer.writeValueAsString(fields)));
+    } catch (Exception e) {
+      logger.warn(marker, format("exception occurred while logging event: %s", event), e);
```

Comment:
I liked the following write-up on logging levels: http://stackoverflow.com/questions/7839565/logging-levels-logback-rule-of-thumb-to-assign-log-levels

For this method, if something weird happened, the overall system is still functional.